docs/design/daemon.md

```diff
@@ -0,0 +1,118 @@
+          app: datastore-shard
+      spec:
+        nodeSelector: 
+          app: datastore-node
+        containers:
+          name: datastore-shard
+          image: kubernetes/sharded
+          ports:
+            - containerPort: 9042
+              name: main
+```
+  - commands that get info
+    - get (e.g. kubectl get daemonsets)
+    - describe
+  - Modifiers
+    - delete (if --cascade=true, then first the client turns down all the pods controlled by the DaemonSet (by setting the nodeName to a non-existant name); then it deletes the DaemonSet; then it deletes the pods)
+    - label
+    - update (only allowed to selector and to nodeSelector and nodeName of pod template)
+    - DaemonSets have labels, so you could, for example, list all DaemonSets with a certain label (the same way you would for a Replication Controller).
+  - In general, for all the supported features like get, describe, update, etc, the DaemonSet works in a similar way to the Replication Controller. However, note that the DaemonSet and the Replication Controller are different constructs.
+
+### Persisting Pods
+  - Ordinary livenes probes specified in the pod template work to keep pods created by a DaemonSet running.
+  - If a daemon pod is killed or stopped, the DaemonSet will create a new replica of the daemon pod on the node.
+
+### Cluster Mutations
+  - When a new node is added to the cluster the DaemonSet starts the daemon on the node (if the node’s labels match the user-specified selectors).
+  - Suppose the user launches a DaemonSet that runs a logging daemon on all nodes labeled “logger=fluentd”. If the user then adds the “logger=fluentd” label to a node (that did not initially have the label), the logging daemon will launch on the node. Additionally, if a user removes the label from a node, the logging daemon on that node will be killed.
+
+## Alternatives Considered
+
+An alternative way to launch daemons is to avoid going through the API server, and instead provide ways to package the daemon into the node. For example, users could:
+
+1. Include the daemon in the machine image
+2. Use static pod manifests to launch daemon pods when the node initializes
+
+These alternatives don’t work as well because the daemons won’t be well integrated into Kubernetes. In particular,
+
+1. In alternatives (1) and (2), health checking for the daemons would need to be re-implemented, or would not exist at all (because the daemons are not run inside pods). In the current proposal, the Kubelet will health-check daemon pods and restart them if necessary.
+2. In alternatives (1) and (2), binding services to a group of daemons is difficult (which is needed in use cases such as the sharded data store use case described above), because the daemons are not run inside pods
+3. A big disadvantage of these methods is that adding new daemons in existing nodes is difficult (for example, if a cluster manager wants to add a logging daemon after a cluster has been deployed).
```

Comment:
BTW, adding is also difficult since other pods may already be scheduled, so that's not a great example.